wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -45,7 +45,13 @@ public final class ZMUserSession: NSObject {
 
     private(set) var coreDataStack: CoreDataStack!
     private let apiServiceFactory: APIServiceFactory
-    private(set) var apiService: APIServiceProtocol?
+    var apiService: APIServiceProtocol? {
+        guard let clientId = selfUserClient?.remoteIdentifier else {
```

Comment:
is this accessed on mainThead?

Reply:
Why?

Reply:
Is the question about the CoreData entity?